composer/core/data_spec.py

```diff
@@ -95,7 +95,8 @@ def __init__(
             self.num_samples = num_samples
 
         else:
-            if isinstance(dataloader.dataset, collections.abc.Sized):
+            # FFCV dataloaders don't have an associated dataset
+            if hasattr(dataloader, "dataset") and isinstance(dataloader.dataset, collections.abc.Sized):
```

Comment:
may want to check with `isinstance` rather than ducktype for dataset and sampler but defer to @ravi-mosaicml here

Reply:
We should probably remove our own `DataLoader` type and the `WrappedDataLoader` (see #424), and instead type the `dataloader` attribute on state with `Iterable[Batch]`. Anything that needs to assume it is a PyTorch dataloader should use   `if isisntance(dataloader, torch.utils.data.DataLoader)` check.

We'll also need to re-implement that `DataloaderProfiler`. Callbacks sholdn't be modifying state to begin with (which it does), and without the WrappedDataLoader, then `isisntance(dataloader, torch.utils.data.DataLoader)` checks will fail once it has been wrapped. Instead, I'm thinking we can add in the `BEFORE_DATALOADER` event to fire immediately before the underlying dataloader iterator is `__next__`'ed, and then restructure the `DataloaderProfiler` to simply measure the time between the before and after calls.

Reply:
Replaced with isinstance and rest will be taken care of in #424